cliquet/views/oauth.py

```diff
@@ -33,15 +37,15 @@ def persist_state(request):
     It will be compared when return from login page on OAuth server.
     """
     state = uuid.uuid4().hex
-    request.registry.session.set(state, request.validated['redirect'])
+    request.registry.session.set(state,
+                                 fxa_conf(request, 'webapp.redirect_url'))
+    request.registry.session.expire(
```

Comment:
Well it is to stay future proof since we probably will need to go for a url whitelist at some point. Having the server linked to only one webclient is a temporary feature. Also we still need to store the state.